### PR TITLE
storage: misc replica comment updates + code movement

### PR DESCRIPTION
**Pull request by knz**

### CODE REVIEW

**Comment by knz** [comment]
fixed the linter error

**Review by tbg — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, 3 of 3 files at r7, 1 of 1 files at r8, 1 of 1 files at r9.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33564#-:-LVjUTn46kCLjab2q86W:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained

___
*[pkg/storage/replica.go, line 2788 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/33564#-LVjSztf1qmgc61ZqvoE:-LVjSztg4T_5P5XnGKYD:bzgj616) ([raw file](https://github.com/cockroachdb/cockroach/blob/02c9a10fda9544a4ac90536a4d77bb10cd44e099/pkg/storage/replica.go#L2788)):*
This isn't quite right, this side effect actually instructs the replica to go into "merge watching mode" (a mode in which it blocks new requests). See `batcheval/cmd_subsume.go`. So a better comment would be

> // A merge is (likely) about to be carried out, and this replica needs to block all traffic until the merge either commits or aborts. See docs/tech-notes/range-merges.md.

`maybeWatchForMerge` probably should be renamed.

Note that https://github.com/cockroachdb/cockroach/pull/33334 hasn't merged yet, so the file isn't there yet. There's also an RFC.
___
*[pkg/storage/replica_gc_queue.go, line 200 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/33564#-LVjUAZB48VNxO1pAC4y:-LVjUAZCB-SDhbv0nNmO:b-l8iqxi) ([raw file](https://github.com/cockroachdb/cockroach/blob/ec2c30bf0b8606f43c535ba3a530928162e582e9/pkg/storage/replica_gc_queue.go#L200)):*
> ```Go
> 
> 	// Now get an updated descriptor for the range.  Note that this may
> 	// not be _our_ range but instead some later range if our range has
> ```

With a merge it would actually be an earlier range. Since the meta entries are keyed by EndKey, and your EndKey becomes the EndKey of the range that merges you, that's the range you'd find.

File: pkg/storage/replica_gc_queue.go [hunk 1]
```diff
@@ -206,10 +210,17 @@ func (rgcq *replicaGCQueue) process(
 		return err
 	}
 	if len(rs) != 1 {
+		// Regardless of whether ranges were merged, we're guaranteed one answer.
+		//
+		// TODO(knz): we should really have a separate type for assertion
```
Comment (nvanbenschoten) [inline pkg/storage/replica_gc_queue.go]: The first of many `TODO(knz)`s in core!

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33564)* status: :shipit: complete! 1 of 0 LGTMs obtained

___
*[pkg/storage/replica.go, line 2788 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/33564#-LVjSztf1qmgc61ZqvoE:-LVnSJFIBtVa8R_v5zVL:b-8l2zwe) ([raw file](https://github.com/cockroachdb/cockroach/blob/02c9a10fda9544a4ac90536a4d77bb10cd44e099/pkg/storage/replica.go#L2788)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

This isn't quite right, this side effect actually instructs the replica to go into "merge watching mode" (a mode in which it blocks new requests). See `batcheval/cmd_subsume.go`. So a better comment would be

> // A merge is (likely) about to be carried out, and this replica needs to block all traffic until the merge either commits or aborts. See docs/tech-notes/range-merges.md.

`maybeWatchForMerge` probably should be renamed.

Note that https://github.com/cockroachdb/cockroach/pull/33334 hasn't merged yet, so the file isn't there yet. There's also an RFC.
</blockquote></details>

I can wait for #33334 to merge.
___
*[pkg/storage/replica_gc_queue.go, line 200 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/33564#-LVjUAZB48VNxO1pAC4y:-LVnSd4gC_Os8RtkKDOL:bqqvtdx) ([raw file](https://github.com/cockroachdb/cockroach/blob/ec2c30bf0b8606f43c535ba3a530928162e582e9/pkg/storage/replica_gc_queue.go#L200)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

With a merge it would actually be an earlier range. Since the meta entries are keyed by EndKey, and your EndKey becomes the EndKey of the range that merges you, that's the range you'd find.
</blockquote></details>

Ack. Thanks for explaining / reminding me.

**Comment by tbg** [comment]
You don't have to wait for that other PR to merge, unless you want to. Still LGTM

**Comment by knz** [comment]
@tbg aiiii I forgot to merge this :crying_cat_face: 

what do we do? do you want to manually resolve this in your branch?

**Comment by tbg** [comment]
I don't have a branch right now, so you could rebase. I won't be touching replica.go today!

**Comment by knz** [comment]
ok, rebased. will merge on green.

**Review by tbg — APPROVED** [review]
Thanks!

**Pull request merged**